src/components/DonationForm/DonationForm.jsx

```diff
@@ -1,54 +1,207 @@
+  //   remove: removeFurniture,
+  // } = useFieldArray({
+  //   control,
+  //   name: 'DonatedFurniture',
+  // });
+
   const {
-    fields: furnitureFields,
-    append: appendFurniture,
-    remove: removeFurniture,
+    fields: descriptionsIntermediateList,
+    update: updateDescription,
+    replace: replaceIntermediateDescriptions,
+    append: appendDescription,
+    remove: removeDescription,
   } = useFieldArray({
     control,
-    name: 'furnitureField',
+    name: 'ImageDetail',
   });
 
+  // const {
+  //   fields: itemsList,
+  //   append: appendItem,
+  //   remove: removeItem,
+  // } = useFieldArray({
+  //   control,
+  //   name: 'Items',
+  // });
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [files, setFiles] = useState([]);
+  const [filesIntermediate, setFilesIntermediate] = useState([]);
+  const [descriptions, setDescriptions] = useState([]);
+
+  const removeIntermediateFile = index => {
+    setFilesIntermediate(
+      filesIntermediate.filter(item => filesIntermediate.indexOf(item) !== index),
+    );
+  };
+
+  const removeIntermediateDescription = index => {
+    removeDescription(index);
+  };
+
+  const getDescription = index => {
+    const descObj = descriptionsIntermediateList[index];
+    if (descObj) {
+      return Object.values(descObj).slice(0, -1).join('');
+    }
+    return '';
+  };
+
+  const updateIntermediateDescription = (index, newDescription) => {
+    if (index >= descriptionsIntermediateList.length) {
+      appendDescription('');
+    }
+    updateDescription(index, newDescription);
+  };
+
+  const onSubmit = async data => {
+    // const urls = await Promise.all(files.map(async file => uploadImage(file)));
+    sendEmail(data.email1, dconfirmemailtemplate);
+  };
+
+  const onOpenModal = e => {
+    setFilesIntermediate(files);
+    replaceIntermediateDescriptions(descriptions);
+    onOpen(e);
+  };
+
+  const onSave = e => {
+    // console.log(descriptionsIntermediateList);
+    setFiles(filesIntermediate);
+    setDescriptions(descriptionsIntermediateList);
+    // console.log(descriptions);
+    onClose(e);
+  };
+
+  const onCancel = e => {
+    // helper function for closing the model
+    setFilesIntermediate(files);
+    replaceIntermediateDescriptions(descriptions);
+    onClose(e);
+  };
+
+  const addDonation = ev => {
+    // appendDonation({ name: ev.target.value, num: 1 });
+    setDonatedFurniture([...DonatedFurnitureList, { name: ev.target.value, num: 1 }]);
+    setFurnitureOptions(prev => prev.filter(e => e !== ev.target.value));
+  };
+
+  const removeDonation = removedName => {
+    console.log(removedName);
+    setFurnitureOptions([...furnitureOptions, removedName]);
+    setDonatedFurniture(prev => prev.filter(e => e.name !== removedName));
+    // removeFurniture({ name: removedName });
+  };
+
+  const onSelectFurniture = async ev => {
+    await addDonation(ev);
+    setSelectedFurnitureValue('Select Furniture');
+  };
```

Comment:
Instead of using an `async` function and `await`ing `addDonation`, we can instead have a `useEffect` with `furnitureOptions` as its dependency, and calling `setSelectedFurnitureValue` there instead. This way, we're using React timings and don't have to worry about the weird interactions between React and pure JS.